### PR TITLE
il: stoploss submitted queue for timeouts

### PR DESCRIPTION
**Pull request by fedekunze**

### CODE REVIEW

File: x/il/keeper/executed_queue.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package keeper
+
+import (
+	"github.com/armon/go-metrics"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k Keeper) TrackPositionTimeout(ctx sdk.Context, currentEthHeight, timeout uint64) {
```
Comment (fedekunze) [inline x/il/keeper/executed_queue.go:9]: @fedekunze @zmanian @jtremback

**Comment by jtremback** [comment]
Like we talked about, let's get clear on the terminology.

Submitted: The outgoing logic call was given to Gravity
|
|
Shroedinger's logic call
|
|
V
Executed: The outgoing logic call changed Ethereum's state and Gravity told you about it

OR

Timed out: You learned that the current eth block is higher than the timeout that was set on the outgoing logic call. This means it can NEVER be executed and it is safe to resubmit

File: x/il/keeper/abci.go [hunk 2]
```diff
@@ -120,19 +130,30 @@ func (k Keeper) EndBlocker(ctx sdk.Context) {
...
 		k.ethBridgeKeeper.SetOutgoingLogicCall(ctx, call)
 
+		// set the submitted value to true and store it
+		stoploss.Submitted = true
+		k.SetStoplossPosition(ctx, address, stoploss)
+
+		pairID := common.HexToAddress(stoploss.UniswapPairID)
+
+		// we now track the pair to recreate it in case of timeout
+		k.SetSubmittedPosition(ctx, timeoutHeight, address, pairID)
```
Comment (marbar3778) [inline x/il/keeper/abci.go:155]: At which point is this pruned?

Comment (marbar3778) [reply x/il/keeper/abci.go:155]: oh in beginblock

File: x/il/keeper/abci.go [hunk 3]
```diff
@@ -152,10 +173,6 @@ func (k Keeper) EndBlocker(ctx sdk.Context) {
 			)
 		}()
 
-		// TODO: technically we should remove the stoploss position now that is has been executed, but we still need to account
-		// for the cases when the outgoing txs to ethereum fail for some reason.
-		k.DeleteStoplossPosition(ctx, address, stoploss.UniswapPairID)
```
Comment (marbar3778) [inline x/il/keeper/abci.go:157]: seems like unless the user deletes the stop loss we wont prune it?

Comment (fedekunze) [reply x/il/keeper/abci.go:157]: correct, this has some tradeoffs. For example, if the IL is decreasing all the time the position will be sitting around for a long time. Ideally, once we implement the hooks on gravity, we can automatically delete the position once we know it was executed on ethereum.

**Pull request merged**